### ISSUE TITLE
Supporting ES6.

### ISSUE THREAD

**Issue by trusktr** [tj/dox#142]
If you could support valid ES6 syntax, that'd be great! For example, dox doesn't work properly with the following:

```js
/*
 * @overview Utility functions used by infamous.
 *
 * LICENSE
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 */

import Engine from 'famous/core/Engine';

/*
 * @function contextWithPerspective
 * Create a context with the given perspective.
 *
 * @param {integer} perspective The amount of perspective to give the context.
 * @return {famous.core.Context} The context with the applied perspective.
 */
export function contextWithPerspective(perspective) {
    var context = Engine.createContext();
    context.setPerspective(perspective);
    return context;
}
```

**Comment by trusktr** [comment]
I can change it to

```js
/*
 * @overview Utility functions used by infamous.
 *
 * LICENSE
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 */

import Engine from 'famous/core/Engine';

/*
 * @function contextWithPerspective
 * Create a context with the given perspective.
 *
 * @param {integer} perspective The amount of perspective to give the context.
 * @return {famous.core.Context} The context with the applied perspective.
 */
function contextWithPerspective(perspective) {
    var context = Engine.createContext();
    context.setPerspective(perspective);
    return context;
}
export contextWithPerspective
```

and that will make it work because the `export` keyword isn't throwing `dox` off.

**Comment by ChiperSoft** [comment]
Can you clarify "doesn't work"? What does dox output that's wrong?

It really shouldn't care at all about the export keyword. The only parsing of source code is done in regexes that match against known function patterns.

**Comment by trusktr** [comment]
@ChiperSoft I saw the regexes, making a PR shortly to show what I mean.

**Comment by trusktr** [comment]
PR: https://github.com/tj/dox/pull/143

**Comment by ChiperSoft** [comment]
Closed with 354c188
